### PR TITLE
GH2099: Also takes care of RFC-02.

### PR DESCRIPTION
**Pull request by tstewart65**
This is the same as pull request 2650 but has been updated to match the changes made in Cake version 1.0.0.

I am hoping that since version 1.0.0 has been released this can be merged.

### CODE REVIEW

File: src/Cake/Commands/DefaultCommandSettings.cs [hunk 1]
```diff
@@ -59,5 +59,9 @@ public sealed class DefaultCommandSettings : CommandSettings
         [CommandOption("--info")]
         [Description("Displays additional information about Cake.")]
         public bool ShowInfo { get; set; }
+
+        [CommandOption("--regeneratecache")]
```
Comment (gitfool) [inline src/Cake/Commands/DefaultCommandSettings.cs:63]: Use `--recompile` instead or as an alias for consistency?

Also, `--regenerate-cache` would be more in line with cli conventions; i.e. separate words with dashes.

Comment (tstewart65) [reply src/Cake/Commands/DefaultCommandSettings.cs:63]: I had --recompile in a previous pull request and they wanted that changed at that time.  That is when I made it --regeneratecache.  If we get a consensus on what is desired I will change it.

Comment (bjorkstromm) [reply src/Cake/Commands/DefaultCommandSettings.cs:63]: I'd say `--invalidate-cache`.

**Comment by pascalberger** [comment]
Closed in favor of #2650. #2650 has been updated with code from this PR.

**Pull request closed without merging**